pkg/resourcegroup/controllers/root/root_controller.go

```diff
@@ -211,6 +212,8 @@ func NewController(mgr manager.Manager, channel chan event.GenericEvent,
 		WithEventFilter(ResourceGroupPredicate{}).
 		// skip the Generic events
 		WithEventFilter(NoGenericEventPredicate{}).
+		// only reconcile resource groups owned by Config Sync
+		WithEventFilter(OwnedByConfigSyncPredicate{}).
```

Comment:
Another option that may be worth looking into is setting up a label filter on the `controller-runtime` client.

The `EventFilter` approach solves the functional requirement of ignoring resource groups not owned by Config Sync, but I assume isn't as efficient compared to filtering resources in the client.

Reply:
IIUC, the client will be configured so that the watch is filtered with the predicates specified like this, as well as any predicates specified to `For` (as functional options); see https://github.com/kubernetes-sigs/controller-runtime/blob/v0.19.0/pkg/builder/controller.go#L321-L323

I don't see a way to specify label filters here without significantly reconfiguring how the controller is initialized. Are you suggesting adding a cache configuration to filter `ResourceGroup`s by labels here? https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/resourcegroup/controllers/runner/run.go#L92-L102 I'm not sure if that's a good idea - I worry that is pretty untested code, so if it's done wrong (now or in the future) it'd risk silently breaking stuff for users...

Reply:
Yes that is what I was suggesting, potentially using label selectors on the caching client to reduce the scope and memory usage of the cache. Agreed that it is risky and prone to gotchas though, so we can punt on it for now. This is a good iterative improvement.